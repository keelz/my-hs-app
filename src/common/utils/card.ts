import CARD from '../constants/card';
import HSJSON from '../constants/hsJson';
import {
  Card,
  CardClassName
} from '../../redux/reducers/Cards';

export const cardSrcWithParameters = (
  id: string,
  locale: string,
  resolution: number,
  ext: string
) => CARD.BASE_URI + CARD.RENDER_PATH
  .replace(`\{${HSJSON.REQUEST_PARAMS.LOCALE}\}`, locale)
  .replace(`\{${HSJSON.REQUEST_PARAMS.RESOLUTION}\}`, resolution.toString())
  .replace(`\{${HSJSON.REQUEST_PARAMS.ID}\}`, id)
  .replace(`\{${HSJSON.REQUEST_PARAMS.EXT}\}`, ext);

export const composeCardClassNames = (cards: Card[]): string[] => cards.reduce(
  (a: string[], b: Card) => {
    const { cardClass } = b;
    switch (cardClass) {
      case CardClassName.DEATHKNIGHT:
      case CardClassName.DREAM:
      case CardClassName.NEUTRAL:
      case CardClassName.WHIZBANG:
      case undefined: return a;
    }
    if (a.indexOf(cardClass) >= 0) return a;
    return [...a, cardClass].sort();
  },
  []
) as string[];

export default {
  cardSrcWithParameters,
  composeCardClassNames,
};
