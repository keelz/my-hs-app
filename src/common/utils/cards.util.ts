import {
  ICard,
  CardClassName
} from '../models/Cards.model';

export const composeCardClassNames = (cards: ICard[]): string[] => {
  // reduce cards to string array and sort.
  const cardClassNames = cards.reduce(
    (a: string[], b: ICard) => {
      const { cardClass } = b;
      switch (cardClass) {
        case CardClassName.NEUTRAL:
        case undefined: return a;
      }
      if (a.indexOf(cardClass) >= 0) return a;
      return [...a, cardClass];
    },
    []
  ).sort() as string[];
  // push neutral type onto array.
  cardClassNames.push(CardClassName.NEUTRAL);
  return cardClassNames;
};

export default {
  composeCardClassNames,
};
