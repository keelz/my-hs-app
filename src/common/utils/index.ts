import * as classnames from 'classnames';

export const composeClassname = (defaultClassname: string | string[]) =>
  (className?: string | string[]) =>
  classnames(defaultClassname, className);

export default {
  composeClassname,
};
