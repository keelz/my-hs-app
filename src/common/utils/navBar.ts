
export const composeTargets = (
  withCardClassNames: string[],
  action: (...args: any[]) => any
) => withCardClassNames.map(cn => ({
  action: () => action(cn),
  path: '/',
  text: cn,
}));
