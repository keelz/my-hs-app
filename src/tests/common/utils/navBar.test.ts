import { composeTargets } from '../../../common/utils/navBar';

const mockCardClassNames = ['one', 'two', 'three'];
const mockAction = jest.fn();

describe('navBar utilities', () => {
  it('composes targets correctly', () => {
    const composedTargets = composeTargets(mockCardClassNames, mockAction);
    composedTargets[0].action && composedTargets[0].action();
    expect(composedTargets).toMatchSnapshot();
    expect(mockAction).toHaveBeenCalled();
  });
});
