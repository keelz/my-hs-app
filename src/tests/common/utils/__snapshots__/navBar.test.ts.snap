// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navBar utilities composes targets correctly 1`] = `
Array [
  Object {
    "action": [Function],
    "path": "/",
    "text": "one",
  },
  Object {
    "action": [Function],
    "path": "/",
    "text": "two",
  },
  Object {
    "action": [Function],
    "path": "/",
    "text": "three",
  },
]
`;
